packages/flutter_tools/lib/src/commands/create.dart

```diff
@@ -135,6 +141,16 @@ abstract class Template {
     });
   }
 
+  bool _hasRelativePath(String path1, String path2) {
```

Comment:
I don't understand this code.
Suppose the two paths are "/foo" and "/bar/baz". There's a simple relative path from one to the other: "bar/baz". But this function will return false unless I'm misunderstanding it.

I suggest exposing this function to tests and testing it with a bunch of cases.